test/framework/cluster/validations/cluster.go

```diff
@@ -360,3 +364,22 @@ func getWorkerNodeMachineSets(ctx context.Context, vc clusterf.StateValidationCo
 	}
 	return ms.Items, nil
 }
+
+func validateKCPobjectForAPIServerExtraArgs(ctx context.Context, vc clusterf.StateValidationConfig) error {
+	kcp, err := controller.GetKubeadmControlPlane(ctx, vc.ManagementClusterClient, vc.ClusterSpec.Cluster)
+	if err != nil {
+		return fmt.Errorf("failed to retrieve kcp: %s", err)
+	}
+	if kcp == nil {
+		return errors.New("KubeadmControlPlane object not found")
+	}
+	apiServerExtraArgsKCP := kcp.Spec.KubeadmConfigSpec.ClusterConfiguration.APIServer.ExtraArgs
+	apiServerExtraArgsSpec := vc.ClusterSpec.Cluster.Spec.ControlPlaneConfiguration.APIServerExtraArgs
+	if apiServerExtraArgsKCP == nil {
+		return fmt.Errorf("kcp object APIServerExtraArgs is nil expected: %v", apiServerExtraArgsSpec)
+	}
+	if !reflect.DeepEqual(apiServerExtraArgsKCP, apiServerExtraArgsSpec) {
```

Comment:
Does this test actually pass? What we specify in our Cluster spec is not the full set of apiserver flags that KCP will return right?